### PR TITLE
Error Reporting: design.md

### PR DESCRIPTION
**Pull request by rputchak**
This commit includes the design.md file for Error Reporting.

### CODE REVIEW

File: design.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+#### Error Reporting Format
+The following is format of the error report, we plan on implementing for the next week
+1. Type of error
+2. Error message
+3. Source location (line number)
+
+####  AST  Changes
+The following type has to be included in the ast.ts for keeping track of the source location to report the errors:
+> export type SourceLocation = {line_number: number}
```
Comment (yalhessi) [inline design.md:22]: Can you add a column too to your source location? I think `Lezer` provides that information at parsing.

Comment (yalhessi) [reply design.md:22]: Where is `SourceLocation` used? How are you carrying around the source location information? It's not clear to me.

**Comment by yalhessi** [comment]
Nice work! I think you have a reasonable approach here, but you probably need to consider two decisions before you move on to implementation:
1- I think the concept of `SourceLocation` here is excellent, but it's not clear to me where that information is calculated or how it carried around during compilation. If I want to throw an error during typechecking some expression, how do I access the source location of this expression?
2- It would be really neat if you can provide an interface for throwing errors that all other teams can use. To use the same example above, if I'm working on typechecking lists, and I want to throw a cool error like the ones you support, what should I do? It's crucial you figure out a reasonable interface so you communicate that information to other teams.

File: design.md (hunk 1 above)
Comment (SruthiPraveenKGeetha) [reply design.md:22]: Thank you for the valuable comments! Based on your feedback and professor's suggestions in the class today, we are proposing the following changes to our design.

1. Yes, we shall add the column number also to the SourceLocation type.
We are proposing to modify the SourceLocation as following to include the column number : 
`export type SourceLocation = { line: number, column: number }
`

2. We created a pull request (https://github.com/ucsd-cse231-s22/chocopy-wasm-compiler-B/pull/22/files#)
with the following changes : 

- SourceLocation is computed with the help of the Lezer tree using "getSourceLocation()" function in parser.ts
We also added annotation changes to parser.ts and type-check.ts. Propagated these changes to other files as required (lower.ts, compiler.ts, repl.ts, runner.ts)
```
 parser.ts:
- Input: Program
- Output: Program

type-check.ts
- Input: Program
- Output: Program<[Type, SourceLocation]>
```

- We added a new file "error_reporting.ts" with implemented interface for ParseError() and skeletal interfaces for other errors.


Please let us know if we have to document these changes in the design.md

**Pull request merged**